src/core/Akka.Persistence.Tests/JournalHealthCheckSpec.cs

```diff
@@ -0,0 +1,132 @@
+                                               akka.actor.serialize-messages = off
+
+                                   """;
+        return TestConfigs.TestSchedulerConfig
+            .WithFallback(Configuration("PersistenceHealthCheckSpec", extraConfig: extraConfig));
+    }
+
+    public JournalHealthCheckSpec(ITestOutputHelper output) : base(HealthCheckConfig(), output)
+    {
+    }
+
+    [Theory]
+    [InlineData(null)] // default plugin
+    [InlineData("akka.persistence.journal.inmem")]
+    public async Task JournalHealthCheck_should_default_to_Healthy(string? pluginId)
+    {
+        using var cts = new CancellationTokenSource(RemainingOrDefault);
+        var pluginHealth = await Extension.CheckJournalHealthAsync(pluginId, cts.Token);
+
+        Assert.Equal(PersistenceHealthStatus.Healthy, pluginHealth.Status);
+        Assert.NotNull(pluginHealth.Description);
+    }
+
+    [Fact]
+    public async Task JournalHealthCheck_should_return_Degraded_when_CircuitBreaker_is_Open()
+    {
+        // Get the journal actor reference
+        var journal = Extension.JournalFor("akka.persistence.journal.failing-open");
+
+        // Trigger a failure to open the circuit breaker
+        var writeMsg = new WriteMessages(new[] { new AtomicWrite(new Persistent("test", 1, "test-pid")) }.ToImmutableList(),
+            TestActor, 1);
+        journal.Tell(writeMsg, TestActor);
+
+        // Advance time to let the write fail and circuit breaker open
+        var testScheduler = (TestScheduler)Sys.Scheduler;
+        testScheduler.Advance(TimeSpan.FromSeconds(2));
+
+        using var cts = new CancellationTokenSource(RemainingOrDefault);
+        var pluginHealth = await Extension.CheckJournalHealthAsync("akka.persistence.journal.failing-open", cts.Token);
+
+        Assert.Equal(PersistenceHealthStatus.Degraded, pluginHealth.Status);
+        Assert.Contains("Circuit breaker is open", pluginHealth.Description);
+    }
+
+    [Fact]
+    public async Task JournalHealthCheck_should_return_Degraded_when_CircuitBreaker_is_HalfOpen()
+    {
+        // Get the journal actor reference
+        var journal = Extension.JournalFor("akka.persistence.journal.failing-half-open");
+
+        // Trigger a failure to open the circuit breaker
+        var writeMsg = new WriteMessages(new[] { new AtomicWrite(new Persistent("test", 1, "test-pid")) }.ToImmutableList(),
+            TestActor, 1);
+        journal.Tell(writeMsg, TestActor);
+
+        var testScheduler = (TestScheduler)Sys.Scheduler;
+
+        // Advance time past call-timeout to let the write fail and circuit breaker open
+        testScheduler.Advance(TimeSpan.FromSeconds(1));
+
+        // Give the async operations time to complete
+        await Task.Delay(100);
+
+        // Advance time past reset-timeout to transition to half-open
+        testScheduler.Advance(TimeSpan.FromSeconds(1));
+
+        // Give the transition time to complete
+        await Task.Delay(100);
```

Comment:
Even though we can use the `TestScheduler` to advance time, the circuit breaker still has to asynchronously perform its update.